### PR TITLE
[extractor/tele5] Fix Extractor

### LINKED ISSUE #3051: [tele5] Two videos available but only the first video can be downloaded
### Checklist

- [X] I'm asking a question and **not** reporting a bug/feature request
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions including closed ones

### Question

The following page : https://tele5.de/mediathek/the-war-of-the-worlds-krieg-der-welten-teil-1-2/ includes two videos. 
Both videos are available online. The first video can be downloaded via 
yt-dlp -f videoID+audioID .... 
Both videos are not DRM copy protected. 
I have no idea how to get the second video, only one https-side is available.

### Verbose log

```shell
yt-dlp -F -v -o "%(title)s.%(ext)s" "https://tele5.de/mediathek/the-war-of-the-worlds-krieg-der-welten-teil-1-2/"
[debug] Command-line config: ['-F', '-v', '-o', '%(title)s.%(ext)s', 'https://tele5.de/mediathek/the-war-of-the-worlds-krieg-der-welten-teil-1-2/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.1 [c0c2c57d3]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.8.10 (CPython 64bit) - Linux-5.13.0-35-generic-x86_64-with-glibc2.29
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
[debug] Optional libraries: brotli, Cryptodome, secretstorage, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] Using fake IP 53.188.250.251 (DE) as X-Forwarded-For
[debug] [Tele5] Extracting URL: https://tele5.de/mediathek/the-war-of-the-worlds-krieg-der-welten-teil-1-2/
[Tele5] the-war-of-the-worlds-krieg-der-welten-teil-1-2: Downloading webpage
[Tele5] 1843831: Downloading token
[Tele5] 1843831: Downloading JSON metadata
[Tele5] 1843831: Downloading JSON metadata
[Tele5] 1843831: Downloading MPD manifest
[Tele5] 1843831: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1843831:
ID                      EXT RESOLUTION FPS │   FILESIZE   TBR PROTO  │ VCODEC        VBR ACODEC      ABR     ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
hls-160000mp4a.40.2-eng mp4 audio only     │                  m3u8_n │ audio only        unknown                 [eng] eng
hls-64000mp4a.40.2-eng  mp4 audio only     │                  m3u8_n │ audio only        unknown                 [eng] eng
hls-64                  mp4 audio only     │ ~ 40.24MiB   64k m3u8_n │ audio only        mp4a.40.2   64k
dash-994506352          m4a audio only     │ ~ 40.24MiB   64k dash   │ audio only        mp4a.40.2   64k 48000Hz [eng] DASH audio, m4a_dash
dash-995756510          m4a audio only     │ ~100.60MiB  160k dash   │ audio only        mp4a.40.2  160k 48000Hz [eng] DASH audio, m4a_dash
dash-141335910          mp4 320x180     25 │ ~106.36MiB  169k dash   │ avc1.42C015  169k video only              DASH video, mp4_dash
hls-233                 mp4 320x180     25 │ ~146.60MiB  233k m3u8_n │ avc1.42C015  233k video only
dash-1791633737         mp4 480x270     25 │ ~249.88MiB  397k dash   │ avc1.42C01E  397k video only              DASH video, mp4_dash
hls-461                 mp4 480x270     25 │ ~290.12MiB  461k m3u8_n │ avc1.42C01E  461k video only
dash-938469724          mp4 640x360     25 │ ~496.42MiB  789k dash   │ avc1.4D401F  789k video only              DASH video, mp4_dash
hls-853                 mp4 640x360     25 │ ~536.66MiB  853k m3u8_n │ avc1.4D401F  853k video only
dash-805799613          mp4 960x540     25 │ ~991.74MiB 1577k dash   │ avc1.4D401F 1577k video only              DASH video, mp4_dash
hls-1737                mp4 960x540     25 │ ~  1.07GiB 1737k m3u8_n │ avc1.4D401F 1737k video only
dash-1034749509         mp4 1280x720    25 │ ~  1.94GiB 3155k dash   │ avc1.64001F 3155k video only              DASH video, mp4_dash
hls-3315                mp4 1280x720    25 │ ~  2.04GiB 3315k m3u8_n │ avc1.64001F 3315k video only
dash-1024744653         mp4 1920x1080   25 │ ~  3.88GiB 6316k dash   │ avc1.640029 6316k video only              DASH video, mp4_dash
hls-6476                mp4 1920x1080   25 │ ~  3.98GiB 6476k m3u8_n │ avc1.640029 6476k video only
```

### LINKED ISSUE #7955: tele5 - new video player "hyogaplayer.com" not supported - not DRM issue
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Germany

### Provide a description that is worded well enough to be understood

tele5 is using a new video player "hyogaplayer.com", which is not recognized by yt-dlp extractor.

The videos are NOT DRM protected. Tested by switching off the DRM support in Firefox (Settings -> General -> Play DRM-controlled content -> off). Video is played without problems in browser.

Example video: https://tele5.de/mediathek/star-trek-enterprise/in-guter-hoffnung

Tested for the following series (doesn't represent all available series):

- Star Trek - Enterprise (https://tele5.de/mediathek/star-trek-enterprise/videos)
- Star Trek - Das nächste Jahrhundert (https://tele5.de/mediathek/star-trek-das-naechste-jahrhundert/videos)
- Star Trek - Raumschiff Enterprise (https://tele5.de/mediathek/raumschiff-enterprise/videos)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://tele5.de/mediathek/star-trek-enterprise/aufbruch-ins-unbekannte-teil-1']
[debug] User config "/home/REDACTED/.config/yt-dlp/config": ['-o', '~/REDACTED/%(title)s.%(ext)s']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481]
[debug] Python 3.11.3 (CPython x86_64 64bit) - Linux-6.1.44-1-MANJARO-x86_64-with-glibc2.38 (OpenSSL 3.1.2 1 Aug 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.0 (setts), ffprobe 6.0, rtmpdump 2.4
[debug] Optional libraries: certifi-2023.07.22, mutagen-1.46.0, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
yt-dlp is up to date (stable@2023.07.06)
[debug] Using fake IP 53.151.23.150 (DE) as X-Forwarded-For
[Tele5] Extracting URL: https://tele5.de/mediathek/star-trek-enterprise/aufbruch-ins-unbekannte-teil-1
[Tele5] aufbruch-ins-unbekannte-teil-1: Downloading webpage
ERROR: [Tele5] aufbruch-ins-unbekannte-teil-1: Unable to extract video player; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 710, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/tele5.py", line 76, in _real_extract
    player_element = self._search_regex(r'(<hyoga-player\b[^>]+?>)', webpage, 'video player')
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 1258, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
```

### PR DESCRIPTION
**Pull request by menschel**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

[extractor/tele5] Fix Extractor
  tele5 now uses loma-cms as middle-ware / interface

[extractor/dplay] Introduce api v3
  new v3 does use json based approach
  old does use path based approach
  keep old behaviour standard

Fixes #7955
Fixes #3051

Merging Video with Audio requires a recent ffmpeg.

To be clear about this PR, it goes to yt-dlp because yt-dl PR-process is unreliable and I do have enough of professional attitude to lay personal differences with your project's main actors aside for a while.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🆕🔄💅

This pull request enhances the extraction of videos from Tele5 and other websites that use the DPlay platform. It adapts the `DPlayIE` and `Tele5IE` extractors to the new v3 api and cms versions and consolidates them in the `dplay.py` file. It also improves the code style and the test cases.

> _Oh we're the coders of the `DPlayIE` class_
> _And we work with the v3 api at last_
> _We heave and we haul and we fix the `Tele5IE`_
> _On the count of three, one, two, three, heave away!_

### Walkthrough
*  Move `Tele5IE` class from `tele5.py` to `dplay.py` and add imports and helper functions for new extraction logic ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-da311cd97bc5b5adb4cde1a3f29fe5bbb7e32ccfcfedeceee45f1d86e3f974aeL1-R68), [link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3R4-R7))
*  Modify `_update_disco_api_headers` method of `DPlayIE` class to handle different authorization headers depending on api version ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3L52-R110))
*  Modify `_get_disco_api_info` method of `DPlayIE` class to handle different referer and origin headers depending on api version ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3L74-R128))
*  Modify `_get_disco_api_info` method of `DPlayIE` class to pass api version to `_download_video_playback_info` method ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3L100-R151))
*  Modify `_real_extract` method of `Tele5IE` class to use new v3 and loma-cms api for Tele5 website ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-da311cd97bc5b5adb4cde1a3f29fe5bbb7e32ccfcfedeceee45f1d86e3f974aeL74-R170))
*  Add `only_matching` key to second and third test cases for `Tele5IE` class ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-da311cd97bc5b5adb4cde1a3f29fe5bbb7e32ccfcfedeceee45f1d86e3f974aeR83), [link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-da311cd97bc5b5adb4cde1a3f29fe5bbb7e32ccfcfedeceee45f1d86e3f974aeR98))
*  Break long lines of tags and note in test case for `DiscoveryNetworksDeIE` class and `_entries` method of `DiscoveryPlusShowBaseIE` class ([link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3L897-R949), [link](https://github.com/yt-dlp/yt-dlp/pull/8501/files?diff=unified&w=0#diff-173ece23309d80df72e1cd7959656991de7efd45236d689a7556e423a84c40b3L947-R1000))

</details>

### CODE REVIEW

File: .github/workflows/quick-test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       with:
         python-version: '3.11'
     - name: Install test requirements
-      run: pip install pytest pycryptodomex
+      run: pip install -r requirements.txt
```
Comment (pukkandan) [inline .github/workflows/quick-test.yml:18]:
```suggestion
      run: pip install pytest pycryptodomex
```

**Comment by geosone** [comment]
would be great if the tele5 extractor can handle the all videos link of the mediathek like

https://tele5.de/mediathek/relic-hunter-die-schatzjaegerin/videos
or
https://tele5.de/mediathek/colony/videos

so it would download all the listed videos.

**Comment by JerryZhouSirui** [comment]
Hi we are a team from University of Michigan's software engineering course and we are willing to work on this issue for our course project. Could we be assigned to this task?

**Comment by bashonly** [comment]
@JerryZhouSirui please choose a different issue to work on

File: yt_dlp/extractor/dplay.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import json
 import uuid
 
+from urllib.parse import urlsplit, urljoin
+
+import requests
```
Comment (pukkandan) [inline yt_dlp/extractor/dplay.py:6]: Do not use requests. All network access should go through the helper functions

File: yt_dlp/extractor/tele5.py [hunk 3]
```diff
@@ -71,18 +184,64 @@ class Tele5IE(DPlayIE):  # XXX: Do not subclass from concrete IE
...
-            endpoint = '%s-%s' % (source_type, endpoint)
-        try:
-            return self._get_disco_api_info(url, asset_id, endpoint, realm, country)
-        except ExtractorError as e:
-            if getattr(e, 'message', '') == 'Missing deviceId in context':
-                self.report_drm(video_id)
-            raise
+        content_regex = re.compile(
+            r'https?://(?:www\.)?(?P<environment>[^.]+)\.de/(?P<parent_slug>[^/]+)/(?P<slug>[^/?#&]+)')
+        m = content_regex.search(url)
```
Comment (pukkandan) [inline yt_dlp/extractor/tele5.py:189]: Read contributing guidelines for helper functions like `_search_regex`, `traverse_obj` etc

**Review by pukkandan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**